lib/apple/api/charts.rb

```diff
@@ -7,18 +7,20 @@ module Charts
 
       def charts(**args)
         res = perform_get("/v1/catalog/#{get_storefront}/charts/", **args)
-        if res && res[:results]
-          res[:results].map do |type, charts|
-            charts.map do |chart|
-              {
-                chart_name: chart[:name],
-                type: type.to_s,
-                chart: chart[:chart],
-                adams: generate_objects(type, chart[:data]),
-              }
-            end
-          end.flatten
-        end
+        return [] unless res && res[:results]
+
+        res[:results].map do |type, charts|
+          return [] if charts.nil?
```

Comment:
`return` will blast you out of the whole calling method, but I'm not sure that's what you want. If you just want to skip, then use `next [] if charts.nil?`

Reply:
good catch @karmiclychee